### DISCUSSION TITLE
Seq2seq model gets error 'hook 'deeplift_grad' has changed the size of value'

### DISCUSSION THREAD

**Discussion by Sprite101** [shap/shap#3361] [Q&A]
This is my model. It's in Pytorch.
custom_CNN_Seq2Seq(
  (attentions): CNN(
    (features): Sequential(
      (0): Conv2d(22, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU()
      (3): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (4): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (5): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (6): ReLU()
      (7): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (8): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (9): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (10): ReLU()
      (11): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (fc): Sequential(
      (0): Linear(in_features=9216, out_features=4098, bias=True)
      (1): LeakyReLU(negative_slope=0.2)
      (2): Dropout(p=0.5, inplace=False)
      (3): Linear(in_features=4098, out_features=2048, bias=True)
      (4): LeakyReLU(negative_slope=0.2)
      (5): Dropout(p=0.5, inplace=False)
      (6): Linear(in_features=2048, out_features=1024, bias=True)
      (7): LeakyReLU(negative_slope=0.2)
      (8): Dropout(p=0.5, inplace=False)
      (9): Linear(in_features=1024, out_features=512, bias=True)
      (10): LeakyReLU(negative_slope=0.2)
      (11): Dropout(p=0.5, inplace=False)
      (12): Linear(in_features=512, out_features=128, bias=True)
    )
  )
  (encoder): Encoder(
    (gru): GRU(128, 256, batch_first=True)
  )
  (decoder): Decoder(
    (attn): LuongAttn()
    (gru): GRU(1, 256)
    (concat): Sequential(
      (0): Linear(in_features=512, out_features=256, bias=False)
      (1): LeakyReLU(negative_slope=0.2)
    )
    (output): Sequential(
      (0): Linear(in_features=256, out_features=1, bias=True)
    )
  )

And I'm getting  this message:
Traceback (most recent call last):
  File "D:\CNN_LSTM\Explainer\CNN_Seq2seq_SHAP\SHAP\SHAP.py", line 91, in <module>
    m = shap_explainer(model, model_path, background_data, explain_data)
  File "D:\CNN_LSTM\Explainer\CNN_Seq2seq_SHAP\SHAP\SHAP.py", line 70, in shap_explainer
    shap_values = e.shap_values(test_input)
  File "D:\anaconda\lib\site-packages\shap\explainers\_deep\__init__.py", line 125, in shap_values
    return self.explainer.shap_values(X, ranked_outputs, output_rank_order, check_additivity=check_additivity)
  File "D:\anaconda\lib\site-packages\shap\explainers\_deep\deep_pytorch.py", line 193, in shap_values
    sample_phis = self.gradient(feature_ind, joint_x)
  File "D:\anaconda\lib\site-packages\shap\explainers\_deep\deep_pytorch.py", line 125, in gradient
    grad = torch.autograd.grad(selected, x,
  File "D:\anaconda\lib\site-packages\torch\autograd\__init__.py", line 276, in grad
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: hook 'deeplift_grad' has changed the size of value

**Comment by CloseChoice** [comment]
@Sprite101 thanks for the report. Could you please open an issue with your report? Please proved a minimal reproducible example aswell, so the definition of the model and the data should be in the snippet.
